edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeOperator.cs

```diff
@@ -922,13 +919,16 @@ private async Task GetOrCreatePersistentVolumeClaim(string name, bool readOnly)
                 }
             };
 
-            if (this.persistentVolumeNameIsSc)
+            // prefer persistent volume name to storage class name, of both are set.
+            if (this.storageClassName.HasValue && this.persistentVolumeName.HasValue)
             {
-                persistentVolumeClaimSpec.StorageClassName = this.persistentVolumeName;
+                Events.DefaultToPvc();
+                this.persistentVolumeName.ForEach(pvName => persistentVolumeClaimSpec.VolumeName = pvName);
             }
             else
             {
-                persistentVolumeClaimSpec.VolumeName = this.persistentVolumeName;
+                this.persistentVolumeName.ForEach(pvName => persistentVolumeClaimSpec.VolumeName = pvName);
+                this.storageClassName.ForEach(scName => persistentVolumeClaimSpec.StorageClassName = scName);
```

Comment:
Should we be setting both SC and PV name here? Maybe this `if` block should look like this?
```csharp
if (this.persistentVolumeName.HasValue)
{
  if (this.storageClassName.HasValue)
  {
    Events.DefaultToPvc();
  }
  this.persistentVolumeName.ForEach(pvName => persistentVolumeClaimSpec.VolumeName = pvName);
}
else if (this.storageClassName.HasValue)
{
  this.storageClassName.ForEach(scName => persistentVolumeClaimSpec.StorageClassName = scName);
}
```

Reply:
Functionally, it's the same, but that's easier to understand.

Reply:
You're right. I didn't parse the logic properly!